src/vm/runners/cairo_runner.rs

```diff
@@ -1022,6 +1022,19 @@ impl CairoRunner {
 
         Ok(())
     }
+
+    /// Add (or replace if already present) a custom hash builtin.
+    pub fn add_additional_hash_builtin(&self, vm: &mut VirtualMachine) {
+        // Remove the custom hash runner if it was already present.
+        vm.builtin_runners
+            .retain(|(name, _)| name != "hash_builtin");
+
```

Comment:
If we were to mimic the Python implementation's behaviour, probably yes. However, since I'm not sure if `add_additional_hash_builtin()` is called at most once, or may be called multiple times I can't say 100% sure one thing or another.

For reference, the Python implementation does the following:
`runner.builtin_runners["hash_builtin"] = hash_builtin`

Reply:
We could probably modify in place.
Regardless, `add_additional_*` sounds to me like we accumulate them. `replace_*` sounds more appropriate.